### PR TITLE
Refactor harmony import and harmony accept import dependency to es6

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

current tests should pass

**If relevant, link to documentation update:**

N/A

**Summary**

upgrade HarmonyImportDependency and HarmonyAcceptImportDependency to es6

**Does this PR introduce a breaking change?**

No

**Other Information**

amend usage of HarmonyAcceptImportDependency in HarmonyAcceptDependency

### CODE REVIEW

File: lib/dependencies/HarmonyAcceptDependency.js [hunk 1]
```diff
@@ -21,8 +21,9 @@ class HarmonyAcceptDependency extends NullDependency {
 
 HarmonyAcceptDependency.Template = class HarmonyAcceptDependencyTemplate {
 	apply(dep, source, outputOptions, requestShortener) {
+		const harmonyImportDependencyTemplate = new HarmonyImportDependency.Template();
```
Comment (sokra) [inline lib/dependencies/HarmonyAcceptDependency.js]: That looks weird.

File: lib/dependencies/HarmonyImportDependency.js [hunk 2]
```diff
@@ -2,55 +2,70 @@
...
-		if(!(dep.module.meta && dep.module.meta.harmonyModule)) {
-			content += "/* harmony import */ " + declaration + dep.importedVar + "_default = __webpack_require__.n(" + dep.importedVar + ");" + newline;
+HarmonyImportDependency.Template = class HarmonyImportDependencyTemplate {
+	apply(dep, source, outputOptions, requestShortener) {
+		const content = this.getContent(true, dep, outputOptions, requestShortener);
+		source.replace(dep.range[0], dep.range[1] - 1, "");
+		source.insert(-1, content);
+	}
+
+	getContent(declare, dep, outputOptions, requestShortener) {
```
Comment (sokra) [inline lib/dependencies/HarmonyImportDependency.js]: You could move it into a separate file. `makeImportStatement` and use this function from these places

File: lib/dependencies/HarmonyAcceptDependency.js (hunk 1 above)
Comment (timse) [inline lib/dependencies/HarmonyAcceptDependency.js]: haha true.

File: lib/dependencies/HarmonyImportDependency.js (hunk 2 above)
Comment (timse) [inline lib/dependencies/HarmonyImportDependency.js]: how about `HarmonyImportDependency.makeImportStatement` ?

**Comment by timse** [comment]
is this going the right direction?

**Comment by timse** [comment]
anything left for me to do here @sokra ?

**Pull request merged**